core/src/main/java/org/apache/carbondata/core/carbon/datastore/chunk/impl/FixedLengthDimensionDataChunk.java

```diff
@@ -77,13 +79,75 @@ public FixedLengthDimensionDataChunk(byte[] dataChunk, DimensionChunkAttributes
       rowId = chunkAttributes.getInvertedIndexesReverse()[rowId];
     }
     int start = rowId * chunkAttributes.getColumnValueSize();
+    int dict = getInt(chunkAttributes.getColumnValueSize(), start);
+    row[columnIndex] = dict;
+    return columnIndex + 1;
+  }
+
+  @Override public int fillConvertedChunkData(ColumnVectorInfo[] vectorInfo, int column,
+      KeyStructureInfo restructuringInfo) {
+    ColumnVectorInfo columnVectorInfo = vectorInfo[column];
+    int offset = columnVectorInfo.offset;
+    int vectorOffset = columnVectorInfo.vectorOffset;
+    int len = columnVectorInfo.size + offset;
+    int[] indexesReverse = chunkAttributes.getInvertedIndexesReverse();
+    int columnValueSize = chunkAttributes.getColumnValueSize();
+    CarbonColumnVector vector = columnVectorInfo.vector;
+    for (int j = offset; j < len; j++) {
+      int start =
+          indexesReverse == null ? j * columnValueSize : indexesReverse[j] * columnValueSize;
+      int dict = getInt(columnValueSize, start);
+      if (columnVectorInfo.directDictionaryGenerator == null) {
+        vector.putInt(vectorOffset++, dict);
+      } else {
+        Object valueFromSurrogate =
+            columnVectorInfo.directDictionaryGenerator.getValueFromSurrogate(dict);
+        if (valueFromSurrogate == null) {
+          vector.putNull(vectorOffset++);
+        } else {
+          vector.putLong(vectorOffset++, (long) valueFromSurrogate);
+        }
+      }
+    }
+    return column + 1;
+  }
+
+  @Override
+  public int fillConvertedChunkData(int[] rowMapping, ColumnVectorInfo[] vectorInfo, int column,
+      KeyStructureInfo restructuringInfo) {
+    ColumnVectorInfo columnVectorInfo = vectorInfo[column];
+    int offset = columnVectorInfo.offset;
+    int vectorOffset = columnVectorInfo.vectorOffset;
+    int len = columnVectorInfo.size + offset;
+    int[] indexesReverse = chunkAttributes.getInvertedIndexesReverse();
+    int columnValueSize = chunkAttributes.getColumnValueSize();
+    CarbonColumnVector vector = columnVectorInfo.vector;
+    for (int j = offset; j < len; j++) {
+      int start = indexesReverse == null ?
+          rowMapping[j] * columnValueSize :indexesReverse[rowMapping[j]] * columnValueSize;
```

Comment:
add space after : 
can we add some notes or docs to show the logic, so readers can understand easily